crates/oxc_ast/src/serialize.rs

```diff
@@ -308,6 +308,21 @@ impl ESTree for RegExpFlagsConverter<'_> {
     }
 }
 
+#[ast_meta]
+#[estree(ts_type = "string", raw_deser = "'todo'")]
+pub struct StringLiteralValue<'a, 'b>(pub &'b StringLiteral<'a>);
+
+impl ESTree for StringLiteralValue<'_, '_> {
+    fn serialize<S: Serializer>(&self, serializer: S) {
+        if self.0.lossy {
+            let raw = self.0.raw.unwrap();
+            JsonSafeString(&raw[1..raw.len() - 1]).serialize(serializer);
```

Comment:
I thought this would work similar to codegen, but this doesn't seem to work.

For example, when js has single quoted with double quote inside:

```js
'\uD800"'
```

this ends up as invalid json:

```txt
{
  "type": "Literal",
  "start": 0,
  "end": 9,
  "value": "\uD800"",  <-- double quote inside double quote
  "raw": "'\\uD800\"'"
}
```

Reply:
That sounds good. I think there is a weird case like this unnecessary escape `'\"'`, so perhaps it requires traversing manually character by character? Anyways, I'll close this in favor of your fix :pray:
I was also not sure about raw deser, so I was mostly giving up. :sweat_smile: